cleverhans/model_abs.py

```diff
@@ -0,0 +1,138 @@
+from abc import ABCMeta
+
+
+class Model(object):
+    """
+    An abstract interface for model wrappers that exposes model symbols
+    needed for making an attack. This abstraction removes the dependency on
+    any specific neural network package (e.g. Keras) from the core
+    code of CleverHans. It can also simplify exposing the hidden features of a
+    model when a specific package does not directly expose them (needed by
+    "features adversaries").
```

Comment:
Could you add a pointer to the paper on this topic for users who are not aware of this line of work? I'm assuming you are referring to this: https://arxiv.org/abs/1511.05122